### PR TITLE
Modify benchmarks

### PR DESCRIPTION
**Pull request by dacorvo**
# What does this PR do?

Modify LLM benchmarks section:

- now evaluates decode throughput only,
- updated numbers for LLama 7b and Llama 13b using AWS Neuron SDK 2.18,
- added Mistralv2

### CODE REVIEW

File: benchmark/text-generation/benchmark.py [hunk 1]
```diff
@@ -9,57 +9,46 @@
...
-    tokenizer.padding_side = "left"
-    # Encode the first input tokens
-    tokens = tokenizer(prompts, return_tensors="pt", padding=True)
-    bootstrap_input_ids = tokens.input_ids
-    # Generate the first set of inputs
-    input_ids, latency = generate(model, bootstrap_input_ids, inc_length)
-    input_length = input_ids.size()[-1]
+
+    def get_input_ids(tokens, batch_size, input_length):
+        return tokens.input_ids[:, :input_length].repeat((batch_size, 1))
```
Comment (michaelbenayoun) [inline benchmark/text-generation/benchmark.py:32]: nit: if `tokens.input_ids` initially has a batch size > 1 the output will have a batch size of `batch_size x original_batch_size`, maybe just select the first batch before the repeat.

File: benchmark/text-generation/mistralv2.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        sequence_length=seq_length,
+        auto_cast_type="bf16",
+        num_cores=NUM_CORES,
+    )
+    with TemporaryDirectory() as tmpdir:
+        model.save_pretrained(tmpdir)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        tokenizer.save_pretrained(tmpdir)
+        json_path = f"{model_name}.json"
+        run(tmpdir, 256, 2048, json_path=json_path)
```
Comment (michaelbenayoun) [inline benchmark/text-generation/mistralv2.py:39]: Maybe add `if __name__ == "__main__"` to not run this code at any import.

File: benchmark/text-generation/benchmark.py [hunk 3]
```diff
@@ -9,57 +9,46 @@
...
+        output_tokens = model.generate(input_ids, do_sample=False, max_new_tokens=max_new_tokens)
     end = time.time()
     return output_tokens, (end - start)
 
 
 def run(model_id, inc_length, max_length, json_path=None):
-    prompts = ["One of my fondest memory"]
+    # Encode the reference prompt
+    with open("./wiki.txt") as f:
+        prompt = f.read()
```
Comment (michaelbenayoun) [inline benchmark/text-generation/benchmark.py:23]: You read a 12k lines document and you tokenize everything at once?

**Review by michaelbenayoun — APPROVED** [review]

File: benchmark/text-generation/benchmark.py (hunk 3 above)
Comment (dacorvo) [reply benchmark/text-generation/benchmark.py:23]: yes, tokenizer is a brute

**Pull request merged**